gax/src/main/java/com/google/api/gax/rpc/CallableFactory.java

```diff
@@ -180,144 +109,74 @@ private BatchingCreateResult(
   }
 
   <RequestT, ResponseT> BatchingCreateResult<RequestT, ResponseT> internalCreate(
-      UnaryCallable<RequestT, ResponseT> transportCallable,
+      UnaryCallable<RequestT, ResponseT> innerCallable,
       BatchingCallSettings<RequestT, ResponseT> batchingCallSettings,
       ClientContext clientContext) {
-    UnaryCallable<RequestT, ResponseT> callable =
-        createBaseCallable(transportCallable, batchingCallSettings, clientContext);
     BatcherFactory<RequestT, ResponseT> batcherFactory =
         new BatcherFactory<>(
             batchingCallSettings.getBatchingDescriptor(),
             batchingCallSettings.getBatchingSettings(),
             clientContext.getExecutor(),
             batchingCallSettings.getFlowController());
-    callable =
+    UnaryCallable<RequestT, ResponseT> callable =
         new BatchingCallable<>(
-            callable, batchingCallSettings.getBatchingDescriptor(), batcherFactory);
-    callable =
-        new EntryPointUnaryCallable<>(
-            callable,
-            transportDescriptor.createDefaultCallContext(),
-            getCallContextEnhancers(clientContext));
+            innerCallable, batchingCallSettings.getBatchingDescriptor(), batcherFactory);
     return new BatchingCreateResult<>(batcherFactory, callable);
   }
 
   /**
+   * Create a paged callable object that represents a paged API method. Designed for use by
+   * generated code.
+   *
+   * @param innerCallable the callable to issue calls
+   * @param pagedCallSettings {@link PagedCallSettings} to configure the paged settings with.
+   * @return {@link UnaryCallable} callable object.
+   */
+  public <RequestT, ResponseT, PagedListResponseT>
+      UnaryCallable<RequestT, PagedListResponseT> asPagedVariant(
```

Comment:
It is an old naming, but can you please change it to smething else? "pagedVariant" sounds strange and is not consistent (in terms of naming) with other methods (there are no other "variants").

Do we need "as"  here and for "asLongRunningOpera".

Also, from a factory one would expect homogeneous method naming, but this class is not like that. Can all methods in this class be named something like:
`batching()` or `asBatching()` / `batchingCallable()` / `asBatchingCallable()` / etc;
`paged()`  or `asPaged()` / `pagedCallable()` / `asPagedCallable()` / etc;
`operation()` or ...
`...`.
?

Reply:
I'll go with your renaming idea.